roles/upload-site/tasks/main.yml

```diff
@@ -94,13 +94,10 @@
   shell: "drush @acsf{{ drush_environment }}.{{ stack }}.{{ acsf_site_name }} {{ item }}"
   with_items:
     # Text fields should replace http(s)://sites.stanford.edu/sitename with
-    # https://acsf_site_name.cardinalsites.stanford.edu. Do nothing for vhosts;
-    # references will point to the on-prem site before cutover, and to the ACSF
-    # site after cutover (note that this is an assumption). Also, we are using
-    # {{ acsf_site_name }} here because we can rely on it existing, regardless
-    # whether there's a vhost.
-    - "sar -y 'http://sites.stanford.edu/{{ inventory_hostname }}/' 'https://{{ acsf_site_name }}{{ stanford_environment }}.cardinalsites.stanford.edu/'"
-    - "sar -y 'https://sites.stanford.edu/{{ inventory_hostname }}/' 'https://{{ acsf_site_name }}{{ stanford_environment }}.cardinalsites.stanford.edu/'"
+    # https://acsf_site_name.cardinalsites.stanford.edu, or
+    # https://vhost.cardinalsites.stanford.edu when one exists.
+    - "sar -y 'http://sites.stanford.edu/{{ inventory_hostname }}/' 'https://{% if vhost %}{{ vhost }}{% else %}{{ acsf_site_name }}{% endif %}{{ stanford_environment }}.cardinalsites.stanford.edu/'"
+    - "sar -y 'https://sites.stanford.edu/{{ inventory_hostname }}/' 'https://{% if vhost %}{{ vhost }}{% else %}{{ acsf_site_name }}{% endif %}{{ stanford_environment }}.cardinalsites.stanford.edu/'"
```

Comment:
Ooh, that `if/else` syntax is new, I like.

Reply:
You can use Jinja2 templating with YAML in Ansible: http://docs.ansible.com/ansible/latest/user_guide/playbooks_templating.html